sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/descriptor/MetricDescriptorTest.java

```diff
@@ -67,116 +68,106 @@ void metricDescriptor_isCompatible() {
+            SourceInfo.fromCurrentStack(),
+            InstrumentDescriptor.create(
+                "Name",
+                "description",
+                "unit",
+                InstrumentType.COUNTER,
+                InstrumentValueType.DOUBLE,
+                Advice.create(Arrays.asList(1.0, 2.0))));
+    assertThat(descriptor1).isEqualTo(descriptor2).hasSameHashCodeAs(descriptor2);
+    // Different name overridden by view is not equal
+    descriptor2 =
+        MetricDescriptor.create(
+            View.builder().setName("bar").build(), SourceInfo.fromCurrentStack(), instrument);
+    assertThat(descriptor1).isNotEqualTo(descriptor2).doesNotHaveSameHashCodeAs(descriptor2);
+    // Different description overridden by view is not equal
+    descriptor2 =
+        MetricDescriptor.create(
+            View.builder().setDescription("foo").build(),
+            SourceInfo.fromCurrentStack(),
+            instrument);
+    assertThat(descriptor1).isNotEqualTo(descriptor2).doesNotHaveSameHashCodeAs(descriptor2);
+    // Different aggregation overridden by view is not equal
+    descriptor2 =
+        MetricDescriptor.create(
+            View.builder().setAggregation(Aggregation.lastValue()).build(),
+            SourceInfo.fromCurrentStack(),
+            instrument);
+    assertThat(descriptor1).isNotEqualTo(descriptor2).doesNotHaveSameHashCodeAs(descriptor2);
+    // Different instrument source name is not equal
+    descriptor2 =
+        MetricDescriptor.create(
+            view,
+            SourceInfo.fromCurrentStack(),
+            InstrumentDescriptor.create(
+                "foo",
+                "description",
+                "unit",
+                InstrumentType.COUNTER,
+                InstrumentValueType.DOUBLE,
+                Advice.empty()));
+    assertThat(descriptor1).isNotEqualTo(descriptor2).doesNotHaveSameHashCodeAs(descriptor2);
+    // Different instrument source description is not equal
+    descriptor2 =
+        MetricDescriptor.create(
+            view,
+            SourceInfo.fromCurrentStack(),
+            InstrumentDescriptor.create(
+                "name",
+                "foo",
+                "unit",
+                InstrumentType.COUNTER,
+                InstrumentValueType.DOUBLE,
+                Advice.empty()));
+    assertThat(descriptor1).isNotEqualTo(descriptor2).doesNotHaveSameHashCodeAs(descriptor2);
+    // Different instrument source unit is not equal
+    descriptor2 =
+        MetricDescriptor.create(
+            view,
+            SourceInfo.fromCurrentStack(),
+            InstrumentDescriptor.create(
+                "name",
+                "description",
+                "foo",
+                InstrumentType.COUNTER,
+                InstrumentValueType.DOUBLE,
+                Advice.empty()));
+    assertThat(descriptor1).isNotEqualTo(descriptor2).doesNotHaveSameHashCodeAs(descriptor2);
+    // Different instrument source type is not equal
```

Comment:
Each of these commented sections seems like it's actually testing for a separate thing. Why not make these into actually separate test methods then?

Reply:
Yeah this about the code style question of whether a single unit test method should test multiple cases or just a single one. (I feel like we've talked about this before 😄)

Personally, I do both depending on the situation, mostly depending on how much setup there is for each case. If the setup is small, then splitting each case out into its own method feels like unnecessary boiler plate on the author and reduced readability. On the other hand, if there's lots of special setup and you have to do things like reset the state of mocks, then it becomes fairly obvious to split out into separate methods. There's plenty of examples of both strategies in this repo. In this particular test, I was refactoring an existing test so stuck with the existing pattern. But even if it was new, I probably would have opted to keep all the not equals cases in a single test.

Reply:
> [...] feels like unnecessary boiler plate on the author and reduced readability. 

Until a test fails, and you have to dive into a line of code from a stack trace instead of quickly getting a meaningful test name that matches a method name.

> I was refactoring an existing test so stuck with the existing pattern

Yeah, I respect the need to limit change range in a single PR.

Reply:
> Until a test fails, and you have to dive into a line of code from a stack trace instead of quickly getting a meaningful test name that matches a method name.

Assertj relieves this with the ability to name / describe assertions. Didn't use it here tho, but probably should in cases where a single test method verifies multiple cases.